superset/db_engine_specs.py

```diff
@@ -707,8 +707,50 @@ def convert_dttm(cls, target_type, dttm):
     def get_table_names(cls, inspector, schema):
         """Need to disregard the schema for Sqlite"""
         return sorted(inspector.get_table_names())
+        
+# Start Dremio Support
+class DremioEngineSpec(BaseEngineSpec):
+    engine = 'dremio'
+    import sqlalchemy_dremio
 
+    @classmethod
+    def get_schema_names(cls, inspector):
+        schemas = [row[0] for row in inspector.engine.execute('SHOW SCHEMAS')
+                   if not row[0].startswith('_')]
+        return schemas
+
+    @classmethod
+    def get_table_names(cls, schema, inspector):
+        sql = f'SELECT TABLE_NAME FROM INFORMATION_SCHEMA."TABLES" WHERE TABLE_SCHEMA = \'{schema}\''
+        schemas = [row[0] for row in inspector.engine.execute(sql)
+                   if not row[0].startswith('_')]
+        return sorted(schemas)
+
+    @classmethod
+    def get_normalized_column_names(cls, cursor_description):
+        columns = cursor_description if cursor_description else []
+        return [cls.normalize_column_name(col[0]) for col in columns]
+    
+    @staticmethod
+    def normalize_column_name(column_name):
```

Comment:
Both `get_normalized_column_names()` and `normalize_column_name()` have been removed from `db_engine_specs` in `master` quite some time ago, so these methods would not do anything. Has this been developed against an old version of Superset?

Reply:
yes, it was done for an old version, I will update that

Reply:
@villebro Can you please tell me the workaround to fix in code?

Reply:
I suggest looking at a similar EngineSpec from master branch and making as few changes as possible. In this case both methods I mentioned above have been made redundant. Regarding your comment in Slack, it seems that Dremio requires column names to be lowercase only? If so take a look at how the Redshift spec lowercases column names by overriding the `mutate_label()` method.

Reply:
Thanks @villebro for the info.